syntax = "proto3";

package classifierpb;

message Target {
  oneof target {
    int32 target_int = 1;
    string target_str = 2;
  }
}

message DatasetEntry {
  repeated float features = 1;
  Target target = 2;
}

message Dataset {
  repeated DatasetEntry entries = 1;
}

message SupportSampleEntry {
  int32 sample_id = 1;
  repeated float features = 2;
  Target target = 3;
}

message SupportSamples {
  repeated SupportSampleEntry entries = 1;
}

message TestSampleEntry {
  int32 sample_id = 1;
  repeated float features = 2;
  Target ground_truth = 3;
}

message TestSamples {
  repeated TestSampleEntry entries = 1;
}

message Likelihood {
  float likelihood = 1;
  Target target = 2;
}

message Likelihoods {
  Likelihood likelihood0 = 1;
  Likelihood likelihood1 = 2;
}

message PredictedSampleEntry {
  int32 sample_id = 1;
  repeated float features = 2;
  Target target = 3;
  
  Likelihoods likelihoods = 4;
}

message PredictedSamples {
  repeated PredictedSampleEntry entries = 1;
}
